skfda/exploratory/visualization/_baseplot.py

```diff
@@ -0,0 +1,94 @@
+"""BasePlot Module.
+
+This module contains the abstract class of which inherit all
+the visualization modules, containing the basic functionality
+common to all of them.
+"""
+
+from abc import ABC, abstractmethod
+from typing import List, Optional, Sequence, Union
+
+import matplotlib.pyplot as plt
+from matplotlib.artist import Artist
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+
+from ._utils import _figure_to_svg
+
+
+class BasePlot(ABC):
+    """
+    BasePlot class.
+
+    Attributes:
+        id_function: list of Artist objects corresponding
+            to every instance of our plot. They will be used to modify
+            the visualization with interactivity and widgets.
+        fig: figure over with the graphs are plotted.
+        axes: sequence of axes where the graphs are plotted.
+    """
+
+    @abstractmethod
+    def __init__(
+        self,
+    ) -> None:
+        self.id_function: List[Artist] = []
+        self.fig: Figure = None
+        self.axes: Sequence[Axes] = []
+
+    @abstractmethod
+    def plot(
+        self,
+    ) -> Figure:
+        """
+        Abstract method used to plot the object and its data.
+
+        Returns:
+            Figure: figure object in which the displays and
+                widgets will be plotted.
+        """
+        pass
+
+    @abstractmethod
+    def num_instances(self) -> int:
```

Comment:
Why not `n_samples`?

Reply:
because depending on the object it can change, for example parametricplot can have two fdata

Reply:
And how the interactivity works in that case?

Reply:
as it uses the fdata final there is no problem, the thing is that when we use a baseplot object in multiple display
i need a generic way for n_samples